app/models/ems_event.rb

```diff
@@ -58,14 +58,46 @@ def self.group_name(group)
     group[:name]
   end
 
-  def self.add_queue(meth, ems_id, event)
-    MiqQueue.submit_job(
-      :service     => "event",
-      :target_id   => ems_id,
-      :class_name  => "EmsEvent",
-      :method_name => meth,
-      :args        => [event],
-    )
+  def self.artemis_client
+    require "manageiq-messaging"
+    # FIXME: logger should be set at a global level
+    ManageIQ::Messaging.logger = _log
+    @artemis_client ||= begin
```

Comment:
I wonder if this global caching should be done withing ManageIQ::Messaging itself.  Similar to how you can create a singleton connection with, say, Octokit.

Reply:
I totally agree, but maybe we should do this in a follow up?

Reply:
Messaging gem to maintain a singleton? Let's stick with what you have for now. You may want to have multiple connections, for example, one for subscribe, and one for one time publish.